### PR TITLE
1 as client get list of users

### PR DESCRIPTION
**Pull request by AMBuilds**
Working functionalities + tests for:

- Generic error when getting users list
- Receiving empty list of users
- Receiving users list with 3 ids (happy path)

**Probably merge conflicts, take a look**

### CODE REVIEW

File: tests/app/Application/UsersLIst/UsersListServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        $this->userDataSource = Mockery::mock(UserDataSource::class);
+
+        $this->usersListService = new UsersListService($this->userDataSource);
+    }
+
+    /**
+     * @test
+     */
+    public function emptyListOfUsers()
```
Comment (rubeniriso) [inline tests/app/Application/UsersLIst/UsersListServiceTest.php]: Please snake case

File: tests/app/Infrastructure/Controller/GetUsersListControllerTest.php [hunk 2]
```diff
@@ -30,13 +30,76 @@ protected function setUp(): void
...
+            $response = $this->get('/api/users/list');
+
+            $response->assertStatus(Response::HTTP_OK)->assertExactJson(['list' => array()]);
+    }
+
+    /**
+     * @test
+     */
+    public function list_with_3_users_returned(){
+        /*
```
Comment (rubeniriso) [inline tests/app/Infrastructure/Controller/GetUsersListControllerTest.php]: Please remove this comment as we are using the other method anyway.

**Review by rubeniriso — CHANGES_REQUESTED** [review]

File: tests/app/Application/UsersLIst/UsersListServiceTest.php (hunk 1 above)
Comment (AMBuilds) [reply tests/app/Application/UsersLIst/UsersListServiceTest.php]: Done.

File: tests/app/Infrastructure/Controller/GetUsersListControllerTest.php (hunk 2 above)
Comment (AMBuilds) [reply tests/app/Infrastructure/Controller/GetUsersListControllerTest.php]: Done.

**Review by rubeniriso — APPROVED** [review]
Ok to merge

**Pull request merged**